handsontable/src/core.js

```diff
@@ -565,6 +565,8 @@ export default function Core(rootElement, userSettings, rootInstanceSymbol = fal
           if (colDelta) {
             metaManager.createColumn(startColumnPhysicalIndex, amount);
 
+            datamap.refreshDuckSchema();
+
```

Comment:
```suggestion
```
I'd keep the logic that refreshes the schema within the DataMap class. I suggest moving that call to the `createCol` and `removeCol` methods. Allowing refreshing the data schema outside the class can lead to a maintenance nightmare (when everyone starts calling/using this method) 😄 WDYT?